doc/contrib/new_api.md

```diff
@@ -0,0 +1,510 @@
+# PREVIEW: New `libelektra-core` API 2023
+
+## Core API
+
+### Note on names
+
+From the [use cases](/doc/usecases/core/README.md) we know that there is a collection structure which contains elements with a name and value (and metadata).
+
+The current API uses these names for those concepts:
+
+- Collection: `KeySet`
+- Element: `Key`
+- Name of element: —
+- Value of element: —
+
+The API proposed below uses (to keep as much as possible from above):
+
+- Collection: `ElektraKeyset`
+- Element: `ElektraKey`
+- Name of element: `ElektraKeyname`
+- Value of element: `ElektraKeyvalue`
+
+Another (maybe less confusing) option would be:
+
+- Collection: `ElektraSet` or `ElektraMap`
+- Element: `ElektraEntry`
```

Comment:
:+1: This is a huge change but probably a good idea as the "Key" for a key/value/meta was very confusing.